subsystem/kubeapi_test.go

```diff
@@ -1277,4 +1303,45 @@ var _ = Describe("[kube-api]cluster installation", func() {
 		}
 		checkClusterCondition(ctx, clusterKubeName, controllers.ClusterReadyForInstallationCondition, controllers.ClusterNotReadyReason)
 	})
+
+	It("deploy clusterDeployment with invalid clusterImageSet", func() {
+		secretRef := deployLocalObjectSecretIfNeeded(ctx, kubeClient)
+		clusterDeploymentSpec := getDefaultClusterDeploymentSpec(secretRef)
+		clusterDeploymentSpec.Provisioning.ImageSetRef.Name = "invalid"
+		deployClusterDeploymentCRD(ctx, kubeClient, clusterDeploymentSpec)
+		clusterKubeName := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      clusterDeploymentSpec.ClusterName,
+		}
+		checkClusterCondition(ctx, clusterKubeName, controllers.ClusterSpecSyncedCondition, controllers.BackendErrorReason)
+	})
+
+	It("deploy clusterDeployment with missing clusterImageSet", func() {
+		secretRef := deployLocalObjectSecretIfNeeded(ctx, kubeClient)
+		spec := getDefaultClusterDeploymentSpec(secretRef)
+
+		// Create ClusterDeployment
+		err := kubeClient.Create(ctx, &hivev1.ClusterDeployment{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "ClusterDeployment",
+				APIVersion: getAPIVersion(),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: Options.Namespace,
+				Name:      spec.ClusterName,
+			},
+			Spec: *spec,
+		})
+		Expect(err).To(BeNil())
+
+		clusterKubeName := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      spec.ClusterName,
+		}
+		checkClusterCondition(ctx, clusterKubeName, controllers.ClusterSpecSyncedCondition, controllers.BackendErrorReason)
+
```

Comment:
Should it check `(controllers.ClusterReadyForInstallationCondition, controllers.ClusterNotReadyReason)` as well?